### PR TITLE
Assorted macOS patches from wxTrac

### PR DESCRIPTION
**Pull request by vadz**
I'm reasonably sure those are good, but any review would be still welcome, of course.

### CODE REVIEW

**Comment by vadz** [comment]
@csomor I'm going to merge those soon, please let me know if you have any objections.

**Review by discnl — COMMENTED** [review]
[this comment should have gone to a particular commit, the GH view got be fooled and apparently I must select lines in that commit to comment on]

File: src/osx/cocoa/window.mm [hunk 1]
```diff
@@ -2440,14 +2440,12 @@ When dragging the bottom part of the DND sample ("Drag text from here!")
...
-        wxTextCtrl *tc = wxDynamicCast( wxpeer , wxTextCtrl );
-        wxComboBox *cb = wxDynamicCast( wxpeer , wxComboBox );
-        if ( tc )
+        if ( wxTextCtrl *tc = wxDynamicCast( wxpeer , wxTextCtrl ) )
         {
             tc->MarkDirty();
             tc->SendTextUpdatedEventIfAllowed();
         }
-        else if ( cb )
+        else if ( wxComboBox *cb = wxDynamicCast( wxpeer , wxComboBox ) )
```
Comment (discnl) [inline src/osx/cocoa/window.mm]: I know it's close to your heart but does [No assignments in conditional expressions](http://wxwidgets.org/develop/coding-guidelines/#no_assignment_in_if) no longer apply or was an exception made here?

I'm not a fan but it does look slightly better here with regards to distance of declaration and usage of `cb`, without splitting up the `else if` and making it uglier in the process.

Comment (vadz) [reply src/osx/cocoa/window.mm]: I definitely don't consider this rule for initializing variables declared inside the `if`. The style guide needs to be updated, but this is (by far) not the only item which needs to be brought up to date there, so I'm a bit afraid of starting to change it...

Comment (discnl) [reply src/osx/cocoa/window.mm]: Thanks for the explanation. I didn't consider this distinction with initialization and that makes total sense.

As I don't have the burden of knowing where the style guide is wrong (I will try to use recent sources more for style decisions) I will look into changing this later.

**Pull request merged**